src/daft-logical-plan/src/optimization/rules/reorder_joins/join_graph.rs

```diff
@@ -848,12 +879,14 @@ impl JoinGraphBuilder {
             LogicalPlan::Join(Join {
                 left,
                 right,
-                left_on,
-                right_on,
+                on,
                 join_type,
                 ..
-            }) if *join_type == JoinType::Inner && !left_on.is_empty() => {
-                for l in left_on {
+            }) if *join_type == JoinType::Inner
+                && let (left_on, right_on, _) = on.equi_preds()
```

Comment:
Might have non-equality predicates here. We can support this later but for now only reorder equality joins.